sklearn/metrics/_plot/tests/test_cap_curve_display.py

```diff
@@ -0,0 +1,435 @@
+    assert hasattr(cap_display, "ax_"), "The display must have an ax attribute"
+    assert hasattr(cap_display, "figure_"), "The display must have a figure attribute"
+    assert hasattr(
+        cap_display, "y_true_cumulative"
+    ), "The display must have a y_true_cumulative attribute"
+    assert hasattr(
+        cap_display.y_true_cumulative, "shape"
+    ), "y_true_cumulative must have a shape attribute"
+    assert hasattr(
+        cap_display.y_true_cumulative, "dtype"
+    ), "y_true_cumulative must have a dtype attribute"
+    if plot_chance_level:
+        assert (
+            cap_display.chance_level_ is not None
+        ), "Chance level line should be present"
+
+
+@pytest.mark.parametrize(
+    "response_method", ["auto", "predict_proba", "decision_function"]
+)
+@pytest.mark.parametrize("normalize_scale", [True, False])
+@pytest.mark.parametrize("name", ["Logistic Regression", None])
+def test_display_from_estimator_and_from_prediction(
+    pyplot,
+    logistic_regression_model,
+    data_binary,
+    normalize_scale,
+    name,
+    response_method,
+):
+    X, y = data_binary
+    y_pred = logistic_regression_model.decision_function(X)
+
+    cap_display_from_estimator = CAPCurveDisplay.from_estimator(
+        logistic_regression_model,
+        X,
+        y,
+        normalize_scale=normalize_scale,
+        name=name,
+        response_method=response_method,
+    )
+
+    cap_display_from_prediction = CAPCurveDisplay.from_predictions(
+        y,
+        y_pred,
+        normalize_scale=normalize_scale,
+        name=name,
+    )
+
+    np.testing.assert_allclose(
+        cap_display_from_estimator.y_true_cumulative,
+        cap_display_from_prediction.y_true_cumulative,
+        rtol=1e-6,
+        atol=1e-8,
+    )
+
+
+def test_invalid_response_method(pyplot, logistic_regression_model, data_binary):
+    X, y = data_binary
+    pattern = r"response_method must be in:.*"
+
+    with pytest.raises(ValueError, match=pattern):
+        _ = CAPCurveDisplay.from_estimator(
+            logistic_regression_model, X, y, response_method="invalid input"
+        )
+
+
+def test_unfitted_estimator(pyplot, data_binary):
+    pattern = (
+        r"This .* instance is not fitted yet\. Call 'fit' with "
+        r"appropriate arguments before using this estimator\."
+    )
+    with pytest.raises(NotFittedError, match=pattern):
+        X, y = data_binary
+
+        lr_model_nofit = LogisticRegression(max_iter=1000)
+        _ = CAPCurveDisplay.from_estimator(lr_model_nofit, X, y)
+
+
+def test_estimator_has_too_many_classes(pyplot):
+    X, y = make_classification(
+        n_samples=100,
+        n_features=5,
+        n_informative=3,
+        n_classes=3,
+        random_state=42,
+    )
+
+    lr_model_n_classes = LogisticRegression(max_iter=1000)
+    lr_model_n_classes.fit(X, y)
+
+    pattern = (
+        r"Expected 'estimator' to be a binary classifier\. Got \d+ classes instead\."
+    )
```

Comment:
We know the exact expected message in this case since the number of classes is hard coded in the test.

```suggestion
    pattern = re.escape(
        "Expected 'estimator' to be a binary classifier. Got 3 classes instead."
    )
```

You might need to `import re` at the top of this file.